File: python/questions/isTrionic.py
from typing import List

class Solution:
    def isTrionic(self, nums: List[int]) -> bool:
        n = len(nums)
        p = 1
        while p<n and nums[p]>nums[p-1]:
            p += 1
        if p==1 or p==n:
            return False
        while p<n and nums[p]<nums[p-1]:
            p += 1
        if p==n:
            return False
        while p<n and nums[p]>nums[p-1]:
            p += 1
        return p==n

File: cpp/questions/isTrionic.cpp
#include <vector>

using namespace std;

class Solution {
public:
    bool isTrionic(vector<int>& nums) {
        int n = nums.size();
        int p = 1;
        while (p<n && nums[p]>nums[p-1]) {
            p++;
        }
        if (p==1 || p==n) {
            return false;
        }
        while (p<n && nums[p]<nums[p-1]) {
            p++;
        }
        if (p == n) {
            return false;
        }
        while (p<n && nums[p]>nums[p-1]) {
            p++;
        }
        return p==n;
    }
};
